### PR TITLE
dispoasable email validation 

### LINKED ISSUE #2111: Detect disposable email domains during sign ups
**Goal**

Add an ability to recognize the disposable email domains when the user signs up on the FP Platform. 

Reference Github: https://github.com/ivolo/disposable-email-domains

### PR DESCRIPTION
**Pull request by OzkanAbdullahoglu**
### What does this pull request aim to achieve? 💯

Resolves #2111 

_Please be concise and link any related issue(s):_

- [x] I have checked that no one else is working on similar changes.
- [x] I have read the latest [**README**](README.md) and understand the development workflow.
- [x] The name of this branch is: **`feature/<branch_name>`**, or **`hotfix/<branch_name>`** if this is a hotfix, where `<branch_name>` briefly describes the issue(s) resolved and is prefixed with the issue number(s) (e.g. `1127-1130-update-git-branching-model`). The branch is created in a cloned version of this repo, **not a forked repo**.
- [x] This branch is rebased/merged with the **latest staging** (or the **latest production** if it is a hotfix).
- [x] There are no merge conflicts.
- [x] There are no console warnings and errors.
- [x] On the right hand side of this PR, I have linked the appropriate issue(s) under "Issues" and added a project progress under "Projects".
- [x] The title describes the issue(s) being addressed using format: **`<issue_numbers> - <brief_issue_description>`** (e.g. `#1127/#1130 - Update Git Branching Model`)

### CODE REVIEW

**Comment by OzkanAbdullahoglu** [comment]
Just like the rest of the validations. 

chrome_2021-01-01_22-35-21

@rabbi08 @manualzuru

File: client/src/utils/validators.js [hunk 1]
```diff
@@ -10,11 +11,24 @@ const validateTopLevelDomain = (string) => {
   return false;
 };
 
+const validateDisposableEmail = (email) => {
+  for (const wildcard of wildcards) {
```
Comment (BMariscal) [inline client/src/utils/validators.js]: Let's change it to this for a O(1) lookup
```suggestion
const emailDomain = email.split("@")[1];
const wildcardsSet = new Set(wildcards);
return wildcardsSet.has(emailDomain);

```

Comment (OzkanAbdullahoglu) [reply client/src/utils/validators.js]: Thank you for indicating this, so making a new Set() is also O(n), too, you are right about [Set.prototype.has()](https://adrianmejia.com/data-structures-time-complexity-for-beginners-arrays-hashmaps-linked-lists-stacks-queues-tutorial/#Set-Operations-runtime) is O(1) (not worst case) but when I check the performance of both code I found out that current for loop with an includes is faster, can you check that too? 

```
const validateDisposableEmail = (email) => {
  console.time("LOOP");
  for (const wildcard of wildcards) {
    if (email.includes(wildcard)) {
      console.log("includes");
    }
  }
  console.timeEnd("LOOP");
};
```


```
 const validateDisposableEmail = (email) => {
  console.time("SET");
  const emailDomain = email.split("@")[1];
  const wildcardsSet = new Set(wildcards);
  console.log(wildcardsSet.has(emailDomain));
  console.timeEnd("SET");
};
```

Comment (ddveloper) [reply client/src/utils/validators.js]: I guess @BMariscal meant to build the Set outside of the `validateDisposableEmail()` function

Comment (OzkanAbdullahoglu) [reply client/src/utils/validators.js]: > I guess @BMariscal meant to build the Set outside of the `validateDisposableEmail()` function

make sense, I don't know why I stuck to that function to build the Set. Anyway, can you review, please?
@BMariscal @ddveloper

**Review by ddveloper — APPROVED** [review]
LGTM! 🏕️

**Review by Zaydme — APPROVED** [review]
LGTM!

**Pull request merged**